lucille-core/src/main/java/com/kmwllc/lucille/core/Runner.java

```diff
@@ -164,23 +164,26 @@ public static Map<String, TestMessenger> runInTestMode(String configName) throws
 
   private static Map<String, List<Exception>> logValidation(Config config) throws Exception {
     Map<String, List<Exception>> exceptions = validatePipelines(config);
-    if (exceptions.isEmpty()) {
-      log.info("Configuration is valid");
-    } else {
-      StringBuilder message = new StringBuilder("Configuration is invalid. " +
-          "Printing the list of exceptions for each pipeline\n");
-
-      for (Map.Entry<String, List<Exception>> entry : exceptions.entrySet()) {
-        message.append("\tConnector: ").append(entry.getKey()).append("\tError count: ")
-            .append(entry.getValue().size()).append("\n");
-        int i = 1;
-        for (Exception e : entry.getValue()) {
-          message.append("\t\tException ").append(i++).append(": ")
-              .append(e.getMessage()).append("\n");
+
+    for (Map.Entry<String, List<Exception>> entry : exceptions.entrySet()) {
+      if (!entry.getValue().isEmpty()) {
+        StringBuilder message =
+            new StringBuilder("Configuration is invalid. " + "Printing the list of exceptions for each pipeline\n");
+
+        for (Map.Entry<String, List<Exception>> entryInner : exceptions.entrySet()) {
```

Comment:
let's try to implement what we want without nested for loops over the same set

Reply:
done